Feat: #BBB-72 서적 조회 API 전반적으로 수정

platinouss opened the pull request:
## 작업 개요
- [x] 기술 서적 조회 시 서적 이미지도 함께 볼 수 있도록 하기 위해, 커버 이미지 url도 같이 응답해주도록 적용
- [x] 서적 조회 시, isbn 값이 없는 서적은 제외하도록 수정
- [x] NAVER API를 호출하는 Client 패키지 외부로 이동
<!-- 작업에 대한 요약 설명을 남겨주세요. -->

## 전달 사항

<!-- 작업과 관련된 전달 사항을 남겨주세요. -->

## 참고 자료

<!-- 작업 시 참고했던 자료의 출처를 남겨주세요. -->

platinouss committed:
[bbb-72] ✨Feat: 서적 검색 결과 응답 시 커버 이미지 URL도 같이 응답하도록 구성

platinouss committed:
[bbb-72] ⚡️Fix: isbn 값이 null인 서적은 제외하도록 수정

platinouss committed:
[bbb-72] 🖋️Rename: NAVER API 호출 관련 패키지 외부로 이동

platinouss committed:
[bbb-72] ✅Test: 서적 검색 API 테스트 시, 커버 이미지 URL 필드 검증 부분 추가

msjang4 commented on src/main/java/com/bombombom/devs/client/naver/enums/SortType.java:
```diff
@@ -0,0 +1,6 @@
+package com.bombombom.devs.client.naver.enums;
+
+public enum SortType {
+    SIM,
```
이 부분 멘토님께서 말씀하시지 않았었나요?

platinouss replied:
넵 수정했습니다

msjang4 reviewed (approved):
고생하셨습니다!

platinouss committed:
[bbb-72] 🛠️Refactor: NAVER Open API 호출에 사용되는 검색 옵션 Enum 명칭 명확하게 설정

검색 옵션들이 열거된 Enum의 명칭만으로 의미를 파악할 수 있도록 수정했습니다. 이후 실제 Open API를 호출할 때, 해당 enum의 value 값이 전달될 수 있도록 구성했습니다.

sseunghoon commented on src/main/java/com/bombombom/devs/book/controller/dto/BookListResponse.java:
```diff
@@ -14,12 +15,14 @@ public record BookListResponse(
     public static BookListResponse fromResult(SearchBooksResult searchBooksResult) {
         return BookListResponse.builder()
             .booksInfo(searchBooksResult.booksResult().stream()
+                .filter(bookResult -> Objects.nonNull(bookResult.isbn()))
```
네이버 API 에서 isbn이 없이 응답으로 주는 경우가 있는건가요?

platinouss replied:
넵 단일 책이 아닌 여러 권이 뭉쳐진 서적 정보가 결과로 나올 때가 있는데, 해당 서적 정보는 isbn 값이 비었더라구요

sseunghoon reviewed (approved):
고생하셨습니다 👍
